corehq/apps/reports/util.py

```diff
@@ -602,3 +608,46 @@ def _add_new_user_to_HQ_group(session, local_user):
         _sync_tableau_users_with_hq(domain)
         # Sync the TableauUser model with Tableau users on the remote Tableau instance
         _sync_tableau_users_with_remote(domain)
+
+
+# Attaches to the parse_web_users method
+def add_on_tableau_details(domain, web_user_dicts):
+    tableau_users_for_domain = TableauUser.objects.filter(server=TableauServer.objects.get(domain=domain))
```

Comment:
Might be more efficient to filter by username and just fetch the username and role for all users in one query:

Something like:

```python
TableauUser.objects.filter(
    server=TableauServer.objects.get(domain=domain)
).filter(username__in=usernames).values('username', 'role')
```